go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -744,14 +763,14 @@ func (agent *ActionAgent) StopReplicationAndGetStatus(ctx context.Context) (*rep
 	return mysql.SlaveStatusToProto(rs), nil
 }
 
-// PromoteSlave makes the current tablet the master
-func (agent *ActionAgent) PromoteSlave(ctx context.Context) (string, error) {
+// PromoteReplica makes the current tablet the master
+func (agent *ActionAgent) PromoteReplica(ctx context.Context) (replicationPosition string, finalErr error) {
```

Comment:
Is this behaviorally different from the deprecated method, or are they the same other than the name? The way git had chosen to show the diffs, it's hard to tell.

Reply:
The behavior is different because we now write to topo _before_ calling SetReadOnly. We used to do it _after_. But.. looks like I also changed the old RPC to have the new behavior. I can change it back if we want to preserve it. WDYT?

Reply:
Hm yeah I think it makes sense to keep the old RPC with the old behavior. It may have been ok anyway, but it seems safer to be conservative since the switch to a new RPC gives us that option.

Reply:
Done.